pkg/api/auth.go

```diff
@@ -141,15 +141,16 @@ func basicAuthHandler(c *Controller) mux.MiddlewareFunc {
 			if err != nil {
 				panic(err)
 			}
+			defer f.Close()
 
-			for {
```

Comment:
Can you add a test case to prove this path works?
1. Single entry with a newline at end of file
2. Single entry without a newline at end of file
3. Multiple entries with a newline at end of file
4. Multiple entries without a newline at end of file

Reply:
done